lib/Conversation/index.js

```diff
@@ -24,98 +20,89 @@ class Conversation extends Component {
+    this.props.addComment(commentBody);
   }
 
   resolveConversation() {
-    this.props.actions.resolveConversation(this.props.id);
-  }
-
-  activateConversation() {
-    this.props.actions.activateConversation(this.props.id);
-    this.setState({ isActive: true });
-  }
-
-  deactivateConversation() {
-    this.props.actions.deactivateConversation(this.props.id);
-    this.setState({ isActive: false });
+    this.props.resolveConversation(this.props.id);
   }
 
   render() {
     const isActive = this.state.isActive || this.props.isActive;
     const isActiveClass = isActive ? 'is-active' : '';
     const { id, comments, ...rest } = this.props;
-
     return (
-      <BoundaryClickWatcher
-        outsideClickHandler={this.deactivateConversation}
-        initialClickHandler={this.activateConversation}
-      >
-        <div className={`conversation ${isActiveClass}`}>
-          <div className="conversation__resolve">
-            <Button
-              types={['link-default']}
-              clickHandler={this.resolveConversation}
-            >
-              <ResolveIcon />
-              Resolve Conversation
-            </Button>
-          </div>
-
-          <CommentList
-            {...rest}
-            conversationId={id}
-            comments={comments}
-            focusFallback={this.focusFallback}
-          />
-
-          {!isActive &&
-            comments.length > 1 && (
-              <div className="conversation__reply-count">
+      <div className={`conversation ${isActiveClass}`}>
+        {this.props.resolveConversation &&
+          comments.length > 0 && (
+            <div>
+              <div className="conversation__resolve">
                 <Button
-                  types={['size-sm', 'link-default']}
-                  clickHandler={() => {}}
+                  types={['link-default']}
+                  clickHandler={this.resolveConversation}
                 >
-                  {`View ${comments.length - 1} ${pluralize(
-                    'reply',
-                    comments.length - 1
-                  )}`}
+                  <ResolveIcon />
+                  Resolve Conversation
                 </Button>
               </div>
-            )}
 
-          {this.props.userCanComment && (
-            <div className="conversation__foot">
-              <CommentForm
-                onSubmit={this.addComment}
-                author={this.props.user}
+              <CommentList
+                {...rest}
+                conversationId={id}
+                comments={comments}
+                focusFallback={this.focusFallback}
               />
             </div>
           )}
 
-          <input
-            type="text"
-            className="conversation__focus-fallback"
-            ref={focusFallback => {
-              this.focusFallback = focusFallback;
-            }}
-            aria-hidden="true"
-          />
-        </div>
-      </BoundaryClickWatcher>
+        {!isActive &&
```

Comment:
I wonder if we should change the naming of `isActive` to something like `showComments`. The original naming made sense as we were setting the active state internally but now we're just showing the comments or not. What do you think?